boruta/1-click-boruta/script.whizzml

```diff
@@ -0,0 +1,386 @@
+            (log-info "remaining fields: " (map to-names (keys fields)))
+            (log-info "------------------------------------------------")
+            (recur f-rank fields sh-ds-id (+ runs 1)))))))
+
+;; full
+;; Checks whether all the fields are marked as important or unimportant
+;;
+;; Inputs:
+;;   f-rank: (map) Map from field ID to imporance flag
+;;
+;; Output: (boolean) True if all the fields are classified
+;;
+(define (full f-rank)
+  (let (undetermined (filter (lambda (x) (= x 0)) (values f-rank)))
+    (empty? undetermined)))
+
+;; remove-unimportant
+;; Removes the fields marked as unimportant
+;;
+;; Inputs:
+;;   fields: (map) Fields information structure
+;;   f-rank: (map) Map from field ID to importance flag
+;;
+;; Output: (map) Fields information structure without unimportant fields
+;;
+(define (remove-unimportant fields f-rank)
+  (iterate (acc fields id (keys f-rank))
+    (if (< (get f-rank id 1) 0)
+        (dissoc acc id)
+        acc)))
+
+;; select-by-importance
+;;
+;; Filters the fields according to their importance flag
+;;
+;; Inputs:
+;;   f-map: (map) Map from field ID to importance flag
+;;   score: (integer) Importance flag value
+;;   to-names-fn: (function) Function to transform IDs to field names
+;;
+;; Output: (list) List of field names
+(define (select-by-importance f-map score to-names-fn)
+  (let (selected (filter-map (lambda (x y) (= y score)) f-map))
+    (map to-names-fn (keys selected))))
+
+;; group-importance
+;;
+;; Groups the fields according to their importance
+;;
+;; Inputs:
+;;   f-rank: (map) Map from field ID to importance flag
+;;   fields: (map) Fields information structure
+;;   f-names: (list) Fields names
+;;   objective: (string) objective field name
+;;
+;; Output:  (map) Map from categories (important, unimportant or undecided) to
+;;                list of field names
+;;
+(define (group-importance f-rank fields f-names objective)
+  (let (to-names (lambda (x) (get-in fields [x "name"] false))
+        important (map to-names
+                       (keys (filter-map (lambda(x y) (> y 0))
+                                         f-rank)))
+        important (append important objective)
+        undecided (map to-names
+                       (keys (filter-map (lambda(x y) (= y 0))
+                                         f-rank)))
+        unimportant (filter (lambda (x) (not (or (member? x important)
+                                                 (member? x undecided))))
+                            f-names))
+    {"important" important
+     "unimportant" unimportant
+     "undecided" undecided}))
+
+;; filter-map
+;; Filters a map by applying the corresponding filter function to its keys
+;;
+;; Inputs:
+;;   fn: (function) function to be applied to the map keys and/or values
+;;   map: (map) Map to be filtered
+;;
+;; Output: (map) Filtered map
+(define (filter-map fn map)
+    (iterate (out-map {} key (keys map))
+      (if (fn key (get map key false))
```

Comment:
nit, but, in principle, it's impossible that `(get map key)` will fail here.

also, i recommend not using `map` as the name of a local variable, because it shadows the syntactic form `map`, and you cannot use `(map key)` because `map` is syntax.